migrations/lib/importer/category_name_finder.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Migrations::Importer
+  class CategoryNameFinder < UniqueNameFinderBase
+    def initialize(shared_data, min_length: nil, max_length: nil, max_attempts: nil)
+      super(shared_data, min_length: 1, max_length: 50)
+
+      @last_suffixes_by_parent_id = Hash.new { |h, k| h[k] = {} }
+      @truncations_by_parent_id =
+        Hash.new { |h, k| h[k] = ::LruRedux::Cache.new(TRUNCATION_CACHE_SIZE) }
+    end
+
+    def find_available_name(name, parent_id)
+      with_parent_scope(parent_id) { super(name) }
+    end
+
+    private
+
+    def with_parent_scope(parent_id)
+      @parent_id = parent_id
+      @last_suffixes = @last_suffixes_by_parent_id[parent_id]
+      @truncations = @truncations_by_parent_id[parent_id]
+
+      yield
+    ensure
+      @parent_id = @last_suffixes = @truncations = nil
+    end
```

Comment:
This is a workaround because `UniqueNameFinderBase` works with `SimpleSet`, but categories have an additional key (`parent_id`) and thus use a `KeyValueSet`. I chose this workaround instead of raising the complexity of the base class, since this seems to be the only case where it's needed.